### PR TITLE
[FIX] web_widget_numeric_step: Fix _onChange input fields methods disabled by the widget

### PR DESCRIPTION
**Pull request by sergio-teruel**
cc @Tecnativa
ping @Tardo @pedrobaeza 
TT22667

### CODE REVIEW

**Review by legalsylvain — APPROVED** [review]
Thanks. Lgtm. No test.

**Review by pedrobaeza — APPROVED** [review]

**Comment by pedrobaeza** [comment]
I'm now fearing that people click several times very quickly on the buttons + and -, and this triggers for each click an onchange. Can we set a reasonable time capture window for not launching the onchange until passed? I know this is done on website for the number of items in the cart before processing the update.

**Comment by yajo** [comment]
~~That should be done in the widget itself IMHO.~~ :point_left: I'm not stupid, I just didn't check the code :laughing:

**Comment by yajo** [comment]
Yes, you can do that with a deferred/promise (I don't remember which one is preferred for v12) and a timeout.

**Comment by Tardo** [comment]
Ummm i see the problem here... my initial intention wasn't overwrite '_onChange' method, i did it as a totally new one.
Now with these changes we have two 'change' (InputField + NumericStep) event calling the same method.

Please, remove the 'change' event declared by 'InputField' class... and the 'input' event need by changed too to listen the '.input_numeric_step' element (that is the real 'input' element).
In the NumericStep widget the $el var. is related to the group of input+buttons not only for the input.

A solution to not 'flood' the change event can be use: https://underscorejs.org/#throttle

**Comment by pedrobaeza** [comment]
@Tardo, what about the suggestion of @Yajo's about deferred/promise with a timeout? Can you take over this as you are more experienced with JS?

**Comment by Tardo** [comment]
Updated with some changes... finally i prefer use macrotask (setTimeout) instead of microtask (Promises) because in this case we don't need be "fast as possible". Underscore offers us 'debounce' that for me is exactly that we need in this case.

File: web_widget_numeric_step/static/src/js/numeric_step.js [hunk 1]
```diff
@@ -13,20 +13,25 @@ odoo.define('web_widget_numeric_step.field', function (require) {
...
             'change .input_numeric_step': '_onChange',
+            'input .input_numeric_step': '_onInput',
         }),
         supportedFieldTypes: ['float', 'integer'],
 
         // Values in milliseconds used for mouse down smooth speed feature
         DEF_CLICK_DELAY: 400,
         MIN_DELAY: 50,
         SUBSTRACT_DELAY_STEP: 25,
+        //
```
Comment (yajo) [inline web_widget_numeric_step/static/src/js/numeric_step.js]: :wastebasket:

File: web_widget_numeric_step/static/src/js/numeric_step.js [hunk 2]
```diff
@@ -198,10 +213,11 @@ odoo.define('web_widget_numeric_step.field', function (require) {
         /**
          * Sanitize user input value
          *
-         * @param {ChangeEvent} ev
+         * @override
          */
         _onChange: function (ev) {
             ev.target.value = this._sanitizeNumberValue(ev.target.value);
+            this._super.apply(this, arguments);
```
Comment (yajo) [inline web_widget_numeric_step/static/src/js/numeric_step.js]: Possibly not strictly needed, but this is the common pattern:

```suggestion
            return this._super.apply(this, arguments);
```

**Review by yajo — CHANGES_REQUESTED** [review]
Looks neat! Minor details.

**Comment by Tardo** [comment]
Changes done!

**Review by yajo — APPROVED** [review]

**Pull request merged**